terraform/stacks/umccr_data_portal/workflow/main.tf

```diff
@@ -0,0 +1,231 @@
+terraform {
+  required_version = ">= 0.14"
+
+  backend "s3" {
+    bucket         = "umccr-terraform-states"
+    key            = "umccr_portal_workflow_automation/terraform.tfstate"
+    region         = "ap-southeast-2"
+    dynamodb_table = "terraform-state-lock"
+  }
+
+  required_providers {
+    aws = {
+      version = "~> 3.7.0"
+      source = "hashicorp/aws"
+    }
+  }
+}
+
+provider "aws" {
+  region = "ap-southeast-2"
+}
+
+variable "iap_jwt_token" {}
```

Comment:
While I am happy to have all other (non-sensitive) parameters defined in the TF stack, I am not so sure about this one.
There are several reasons:
- most importantly this would make the deployment machine the keeper of the secret, which it should not be
- if I don't have the (current/latest) token locally, I'd need to pull it from somewhere first, right?
- there is a chance that I accidentally push a local token and overwrite the actual/latest production token
- it complicated this stack unnecessarily

Happy to discuss that, especially since we have quite a mixed use of ParameterStore. Maybe we should keep all non-sensitive configuration parameters in ParameterStore (and have that controlled by TF) and keep real secrets in SecretsManager (which is then manually maintained by the admins).

@brainstorm @alexiswl Opinions? Perhaps a topic to quickly discuss in the next DevOps meeting?

Reply:
Yep; happy to go manual for token.

Reply:
Yup, locks in as docs state, tokens out.

Reply:
A little new to this, is there an issue in keeping it as an encrypted ssm parameter?

Reply:
No, the main issue is not keeping it in ParameterStore vs SecretsManager, but more where the secret comes from initially (local vs remote).
With the first version it is pushed into ParameterStore from the (local) deployment machine. Now the stack assumes it is already (securely) in place on AWS side.

Reply:
Encrypted SSM parameter is ok, I suppose. Just that we don't wish to mange this through terraform -- but just let it simply directly manage it through Console UI or CLI or as such. The key reason being is that terraform still store value in its own state file of key-value pair.

Reply:
The secret is stored in plain text in the terraform state in both cases, so the (local) deployment machine has access to it anyway. (solving that is a completely different beast)
However, in the revised/current stack the secret does **not originate** from the client machine (and I believe is not stored in plain text). So a little more secure, but it completely eliminates the issue that the client instance requires the secret as prerequisite for the stack to work, and that different client machines can have different/conflicting secrets.